assets/components/productPage/productPagePeriodForm/productPagePeriodFormActions.js

```diff
@@ -0,0 +1,19 @@
+// @flow
+import { sendTrackingEventsOnClick, type SubscriptionProduct } from 'helpers/subscriptions';
+
+export type Action<P> = { type: 'SET_PERIOD', period: P, scope: string };
+
+function productPagePeriodFormActionsFor<P>(scope: string, product: SubscriptionProduct | null) {
+  function setPeriod(period: string): (dispatch: Function) => Action<P> {
```

Comment:
Dispatch has the type `Dispatch<A>`, where `A` is the action type. This can be imported from `redux`.